packages/ui/src/components/va-split/VaSplit.vue

```diff
@@ -0,0 +1,281 @@
+      type: Array as any as PropType<[SplitLimit, SplitLimit]>,
+      default: () => [0, 0],
+    },
+  },
+
+  emits: useStatefulEmits,
+
+  setup: (props, { emit }) => {
+    const splitPanelsContainer = shallowRef<HTMLElement>()
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const containerSize = ref()
+    const getContainerSize = () => {
+      containerSize.value = props.vertical ? splitPanelsContainer.value?.offsetHeight : splitPanelsContainer.value?.offsetWidth
+    }
+    onMounted(getContainerSize)
+    useResizeObserver([splitPanelsContainer], getContainerSize)
+
+    const convertToPercents = (v: string | number) => {
+      let numberValue = ''
+      let measureValue = ''
+
+      if (isNumber(v)) { return v }
+
+      v.split('')
+        .filter((char) => char && char !== ' ')
+        .forEach((char) => {
+          !isNaN(+char) ? numberValue += char : measureValue += char
+        })
+
+      switch (measureValue) {
+        case '%':
+          return +numberValue
+        case 'px':
+          return (+numberValue / containerSize.value!) * 100
+        case 'rem':
+          return ((+numberValue * 16) / containerSize.value!) * 100
+        case 'any':
+        case '':
+          return 100
+        default:
+          warn('Invalid limits measure!')
+          return 0
+      }
+    }
+    const getPanelMinMax = (v: SplitLimit) => {
+      if (!v || !containerSize.value) { return }
+
+      let minPercents = 0
+      let maxPercents = 100
+
+      if (isString(v) || isNumber(v)) { minPercents = convertToPercents(v) }
+
+      if (Array.isArray(v)) {
+        minPercents = convertToPercents(v[0])
+        maxPercents = convertToPercents(v[1])
+      }
+
+      if (minPercents > maxPercents) {
+        warn(`Min panels size can not be larger than max one! Passed limit: ${v}.`)
+        maxPercents = minPercents
+      }
+
+      return { min: minPercents, max: maxPercents }
+    }
+
+    const startPanelMinMax = computed(() => getPanelMinMax(props.limits[0]))
+    const endPanelMinMax = computed(() => {
+      const result = getPanelMinMax(props.limits[1])
+
+      // here potentially can be a conflict between two checks below but the first one won't break component
+      if (result?.min && startPanelMinMax.value?.min && Math.floor(result.min + startPanelMinMax.value.min) > 100) {
+        warn('The sum of different panels min sizes should be lesser than 100% of the container size!')
+        result.min = 100 - startPanelMinMax.value.min
+      }
+
+      if (result?.min && result?.max &&
+        startPanelMinMax.value?.min && startPanelMinMax.value?.max &&
+        (Math.ceil(result.min + startPanelMinMax.value.max) < 100 ||
+          Math.ceil(result.max + startPanelMinMax.value.min) < 100)) {
+        warn('The sum of different panels min and max sizes should be equal to 100% of the container size!')
+        result.min = 100 - startPanelMinMax.value.max
+      }
+
+      return result
+    })
```

Comment:
what do you think about this:

```ts
    const startPanelLimits = computed(() => getPanelMinMax(props.limits[0]))
    const endPanelLimits = computed(() => getPanelMinMax(props.limits[1]))

    const startPanelMinMax = computed(() => {
      if (startPanelLimits.value?.min + endPanelLimits.value?.min > 100) {
        warn(...)
      }

      return {
        min: startPanelLimits.value?.min,
        max: Math.min(startPanelLimits.value?.max, 100 - endPanelLimits.value?.min),
      }
    }))

    const endPanelMinMax = computed(() => ({
      min: startPanelLimits.value?.min + endPanelLimits.value?.min > 100
        ? 100 - startPanelLimits.value?.min
        : endPanelLimits.value?.min,
      max: Math.min(endPanelLimits.value?.max, 100 - startPanelLimits.value?.min),
    }))
```
and in `getPanelMinMax` we have to check that min/max does not exceed 100